components/message.py

```diff
@@ -1,15 +1,23 @@
 from textwrap import wrap
+from abc import ABC, abstractmethod
 
 from canvas import Canvas, Colour
 from components import Component
 from data import Data
 
 
-class Message(Component):
-    """ Displays any existing messages at the top of the screen.
+class Message(Component, ABC):
```

Comment:
Because `Component` is abstract, `Message` will actually inherit it's abstractness if we don't implement all the required methods. So, we can actually leave `draw_data` out entirely from this file, and not explicitly inherit `ABC`. Because subclasses of `Message` will also be subclasses of `Component`, and `Message` does not provide an implementation of `draw_data`, these subclasses will then be forced to implement `draw_data`.